src/modules/context2d.js

```diff
@@ -2037,12 +2070,24 @@ import { console } from "../libs/console.js";
           oldLineWidth = this.lineWidth;
           this.lineWidth = oldLineWidth * options.scale;
         }
-        this.pdf.text(options.text, tmpRect.x, tmpRect.y, {
-          angle: options.angle,
-          align: textAlign,
-          renderingMode: options.renderingMode,
-          maxWidth: options.maxWidth
-        });
+
+        if (tmpRect.y <= pageHeightMinusMargin) {
```

Comment:
These conditions are not needed, can be replaced with
`var croppedText = this.pdf.splitTextToSize(options.text, options.maxWidth || pageWidthMinusMargin - tmpRect.x)[0];
            this.pdf.text(croppedText, tmpRect.x, tmpRect.y, {
              angle: options.angle,
              align: textAlign,
              renderingMode: options.renderingMode,
              renderMaxWidthOverflow: false`
This is because getPagesByPath is already accounting for margins.
Also this code was falsely accusing out of bonds characters and when this happened, these characters would not be drawn(because the for loop through the pages only iterates once because there only is one page).